cpp/modmesh/buffer/pymod/array_common.hpp

```diff
@@ -41,6 +42,116 @@
 // See more details in the issue: https://github.com/solvcon/modmesh/issues/283
 #include <modmesh/buffer/pymod/SimpleArrayCaster.hpp>
 
+namespace pybind11
+{
+
+namespace detail
+{
+
+template <>
+struct npy_format_descriptor<modmesh::Complex<double>>
+{
+    static constexpr auto name = const_name("complex128");
+    static constexpr int value = npy_api::NPY_CDOUBLE_;
+
+    static pybind11::dtype dtype()
+    {
+        return pybind11::dtype("complex128");
+    }
+
+    static std::string format()
+    {
+        return "=Zd";
```

Comment:
It will be nice if you can leave a comment here.

Reply:
I have leaved a comment here.